packages/collections/_test.pony

```diff
@@ -752,3 +753,158 @@ class iso _TestHeap is UnitTest
+    t = set0011.clone()      //0b0011
+    t.union(set0101)         //0b0101
+    h.assert_eq[U8](t.value(), 0b0111)
+
+    // a.union(b) == b.union(a)
+    t = set0101.clone()      //0b0101
+    t.union(set0011)         //0b0011
+    h.assert_eq[U8](t.value(), 0b0111)
+
+    // Mutable Operator: intersect
+    t = set0011.clone()      //0b0011
+    t.intersect(set0101)     //0b0101
+    h.assert_eq[U8](t.value(), 0b0001)
+
+    // a.intersect(b) == b.intersect(a)
+    t = set0101.clone()      //0b0101
+    t.intersect(set0011)     //0b0011
+    h.assert_eq[U8](t.value(), 0b0001)
+
+    // Mutable Operator: difference
+    t = set0011.clone()              //0b0011
+    t.difference(set0101)            //0b0101
+    h.assert_eq[U8](t.value(), 0b0110)
+
+    // a.difference(b) == b.difference(a) **symetric difference**
+    t = set0101.clone()      //0b0101
+    t.difference(set0011)    //0b0011
+    h.assert_eq[U8](t.value(), 0b0110)
+
+    // Mutable Operator: remove
+    t = set0011.clone()      //0b0011
+    t.remove(set0101)        //0b0101
+    h.assert_eq[U8](t.value(), 0b0010)
+
+    // a.remove(b) != b.remove(a)
+    t = set0101.clone()      //0b0101
+    t.remove(set0011)        //0b0011
+    h.assert_eq[U8](t.value(), 0b0100)
+
+    // Mutable Operator: flip
+    t = set0011.clone()
+    h.assert_true(set0011 == t)                   //  t = 0b0011
+    t.flip(_TestFlagD)                            // flip 0b1000
+                                                  //      ------
+    h.assert_true(set0011 != t)                   //  t = 0b1011
+
+                                                  //  t = 0b1011
+    t.flip(_TestFlagD)                            // flip 0b1000
+                                                  //      ------
+    h.assert_true(set0011 == t)                   //  t = 0b0011
+
+
+    // Mutable Operator: all
+    t.all()
+    h.assert_true(t.apply(_TestFlagA))
+    h.assert_true(t.apply(_TestFlagB))
+    h.assert_true(t.apply(_TestFlagC))
+    h.assert_true(t.apply(_TestFlagD))
+
+    // Mutable Operator: clear
+    t.clear()
+    h.assert_false(t.apply(_TestFlagA))
+    h.assert_false(t.apply(_TestFlagB))
+    h.assert_false(t.apply(_TestFlagC))
+    h.assert_false(t.apply(_TestFlagD))
+
+    // Mutable Operator: set
+    t .> set(_TestFlagA) .> set(_TestFlagD)
+    h.assert_true(t.apply(_TestFlagA))
+    h.assert_false(t.apply(_TestFlagB))
+    h.assert_false(t.apply(_TestFlagC))
+    h.assert_true(t.apply(_TestFlagD))
+
+    // Check for idempotency
+    t .> set(_TestFlagA) .> set(_TestFlagD)
+    h.assert_true(t.apply(_TestFlagA))
+    h.assert_false(t.apply(_TestFlagB))
+    h.assert_false(t.apply(_TestFlagC))
+    h.assert_true(t.apply(_TestFlagD))
+
+    // Mutable Operator: unset
+    t .> unset(_TestFlagD)
+    h.assert_true(t.apply(_TestFlagA))
+    h.assert_false(t.apply(_TestFlagB))
+    h.assert_false(t.apply(_TestFlagC))
+    h.assert_false(t.apply(_TestFlagD))
+
+    // Check for idempotency
```

Comment:
I was skimming just now and got confused. I think this should be "checked unset for idempotency" and ditto corresponding change elsewhere.